packages/mask/src/plugins/Wallet/services/transaction/watcher.ts

```diff
@@ -51,51 +107,132 @@ async function getTransactionReceipt(chainId: ChainId, hash: string) {
     }
 }
 
-async function checkReceipt() {
-    if (timer !== null) {
-        clearTimeout(timer)
-        timer = null
-    }
-
-    const chainId = currentChainIdSettings.value
-    const map = getTransactionMap(chainId)
-    const transactions = [...map.entries()].sort(([, a], [, z]) => z.at - a.at)
-    const watchedTransactions = transactions.slice(0, WATCHED_TRANSACTIONS_SIZE)
-    const unwatchedTransactions = transactions.slice(WATCHED_TRANSACTIONS_SIZE)
-    unwatchedTransactions.forEach(([hash]) => unwatchTransaction(chainId, hash))
-
-    const checkResult = await Promise.allSettled(
-        watchedTransactions.map(async ([hash, transaction]) => {
-            const receipt = await map.get(hash)?.receipt
-            if (receipt) return true
-            map.set(hash, {
-                at: transaction.at,
+async function checkReceipt(chainId: ChainId) {
+    await Promise.allSettled(
+        storage.getWatched(chainId).map(async ([hash, transaction]) => {
+            const receipt = await storage.getItem(chainId, hash)?.receipt
+            if (receipt) return
+            storage.setItem(chainId, hash, {
+                ...transaction,
                 receipt: getTransactionReceipt(chainId, hash),
             })
-            return false
         }),
     )
+}
+
+async function checkAccount(chainId: ChainId, account: string) {
+    const API_URL = resolveExplorerAPI(chainId)
+    const { API_KEYS = [] } = getExplorerConstants(chainId)
+
+    const watchedTransactions = storage.getWatched(chainId)
+    const latestTransactions = await getLatestTransactions(account, API_URL, {
+        offset: CHECK_LATEST_SIZE,
+        apikey: first(API_KEYS),
+    })
+
+    for (const latestTransaction of latestTransactions) {
+        const [watchedHash, watchedTransaction] =
+            watchedTransactions.find(([hash, transaction]) => {
+                // the transaction hash exact matched
+                if (latestTransaction.hash === hash) return true
```

Comment:
If hashes are the same, why we need to call `replaceRecentTransaction()`? Since speed up create a new tx.

Reply:
The latest transaction fetched from explorer may not be a replaced one.